### PR TITLE
Fix file info of new uploaded file not being persisted after edit

### LINKED ISSUE #8097: strapi-provider-upload: file values (name, alt, caption) are not persisted when uploading from a component
**Describe the bug**
When attempting to edit the fields (i.e. name, description, caption) of a new media object (e.g., image) before uploading it, the values are not persisted. This only happens when a media object is uploaded directly via a component (e.g., from inside a Restaurant component which contains a media field). If the media object is uploaded via the Media Library, the values of the fields are correctly persisted and will be passed to the upload function. 

**Steps to reproduce the behavior**
1. Go to any content type of your choice that contains **a media field**
2. Click on an existing entry
3. Press the "Add" Icon on the media field
4. Press the button "Add more assets"
5. Press the button "Browse files" and select a random image
6. Press the "Edit" Icon on the image preview
7. Enter text into the fields **Alternative text, Caption** or change the **File name**
8. Press the button "Finish"
9. Press the "Edit" Icon on the image preview (again)
10. Notice that the text enter previously is missing (i.e. has not been stored)

**Expected behavior**
The changed values are persisted such that they can be provided before uploading the media object. 

**Possibly Affected Files**
The reducer code for the case `"ON_CHANGE"` is different between the following two files: 
- /strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js
- /strapi-plugin-upload/admin/src/containers/ModalStepper/reducer.js

The `InputModalStepperProvider` uses the **lodash** `set` function and a `draftState`: 
`set(draftState.fileToEdit, action.keys.split('.'), action.value);`
The `ModalStepper` calls the function `state.updateIn(['fileToEdit', ...action.keys.split('.')], () => action.value);`.

**Screenshots**
I'm not sure if the screenshot is of much help. But my guess was that it might have something to do with the `state.fileInfo` being updated. But the name of the image remains the same in `state.file.name`.  
image

**System**
- Node.js version: v12.8.4
- NPM version: 6.14.6
- Strapi version: 3.1.6
- Database: Postgres
- Operating system: Linux

**Additional Information**
I would be happy to provide a fix if someone could guide me or someone has an idea what might be the issue. Thanks a lot for the feedback and investigating!

### PR DESCRIPTION
**Pull request by henrych4**
### Description of what you did:

fix #8097 

dispatch an action to update file info after edited a new uploaded file.

### CODE REVIEW

File: packages/strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js [hunk 1]
```diff
@@ -145,6 +145,12 @@ const reducer = (state, action) =>
         draftState.selectedFiles = draftState.selectedFiles.filter(file => file && file.id);
         break;
       }
+      case 'ON_SUBMIT_EDIT_NEW_FILE': {
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js:148]: Can you add the related test for this case?

**Review by soupette — CHANGES_REQUESTED** [review]
Thanks a lot @henrych4!

Can you just add a small test for the reducer's new case?

File: packages/strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js (hunk 1 above)
Comment (henrych4) [reply packages/strapi-plugin-upload/admin/src/containers/InputModalStepperProvider/reducer.js:148]: test added.

**Review by soupette — APPROVED** [review]
LGTM!

Thanks a lot for your contribution.

**Comment by lauriejim** [comment]
This pull request has been mentioned on **Strapi Community**. There might be relevant details there:

https://forum.strapi.io/t/new-release-strapi-v3-2-4-security-fix/509/1

**Pull request merged**